src/wallet/wallet.cpp

```diff
@@ -1708,9 +1709,10 @@ int64_t CWallet::RescanFromTime(int64_t startTime, const WalletRescanReserver& r
  * from or to us. If fUpdate is true, found transactions that already
  * exist in the wallet will be updated.
  *
- * Returns null if scan was successful. Otherwise, if a complete rescan was not
- * possible (due to pruning or corruption), returns pointer to the most recent
- * block that could not be scanned.
+ * Returns true if scan was successful. If a complete rescan was not
+ * possible (due to pruning or corruption), returns false and sets failed_block
+ * to the most recent block that could not be scanned. If the rescan was aborted
+ * before it could complete, returns false and does not set failed_block.
```

Comment:
Not sure if in this PR, but would it make sense to just return an `enum` to better distinct between `fail`, `ok` and `user-aborted`?

Reply:
Could be done here IMO, not a big change and would be a lot better IMO.

Reply:
Updated to return an enum.